packages/core/src/server.ts

```diff
@@ -1872,7 +1890,10 @@ export class Controller extends API {
                 (typeof requestId === "undefined" || r.id === requestId) &&
                 r.token === token &&
                 r.status === AuthRequestStatus.Verified &&
-                r.purpose === purpose
+                (r.purpose === purpose ||
+                    (purpose === AuthPurpose.Signup && r.purpose === AuthPurpose.v3_Signup) ||
+                    (purpose === AuthPurpose.Login && r.purpose === AuthPurpose.v3_Login) ||
+                    (purpose === AuthPurpose.Recover && r.purpose === AuthPurpose.v3_Recover))
```

Comment:
This was necessary because the `authRequest` can come with the original `purpose`, but the `_useAuthToken` will come with the "new" purpose, so I wanted to make sure those would still match.